dynawo/sources/API/CSTR/CSTRJsonExporter.cpp

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright (c) 2015-2025, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  CSTRJsonExporter.cpp
+ *
+ * @brief Dynawo constraint JSON exporter : implementation file
+ *
+ */
+#include <fstream>
+#include <sstream>
+
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include "DYNMacrosMessage.h"
+#include "DYNCommon.h"
+
+#include "CSTRJsonExporter.h"
+#include "CSTRConstraintsCollection.h"
+#include "CSTRConstraint.h"
+
+using std::fstream;
+using std::ostream;
+using std::string;
+
+using boost::property_tree::ptree;
+
+namespace constraints {
+
+void
+JsonExporter::exportToFile(const std::shared_ptr<ConstraintsCollection>& constraints, const string& filePath) const {
+  fstream file;
+  file.open(filePath.c_str(), fstream::out);
+  if (!file.is_open()) {
+    throw DYNError(DYN::Error::API, FileGenerationFailed, filePath.c_str());
+  }
+
+  exportToStream(constraints, file);
+  file.close();
+}
+
+void
+JsonExporter::exportToStream(const std::shared_ptr<ConstraintsCollection>& constraints, ostream& stream, double afterTime) const {
+  ptree root;
+  ptree array;
+  for (const auto& constraintPair : constraints->getConstraintsById()) {
+    const auto& constraint = constraintPair.second;
+    std::cout << afterTime << " ? " << constraint->getTime() << std::endl;
+    if (!DYN::doubleGreater(constraint->getTime(), afterTime))
```

Comment:
This is a costly operation, If there are many constraints the performance might suffer. Wouldn't it be easier to empty the constraint after each dump in TR mode?